tests/unit/multipleAndNestedViewsActivateCalls/Test.js

```diff
@@ -49,296 +44,281 @@ define([
-						"content_P2_P2S1_P2V2");
-					multipleAndNestedViewsActivateCallsApp1P2S1View = viewUtils.getViewFromViewId(testApp,
-						"content_P2_P2S1");
-
-					//	checkNestedNodeVisibility(multipleAndNestedViewsActivateCallsApp1Content,
-					// multipleAndNestedViewsActivateCallsApp1P2V2);
-					assert.isNull(multipleAndNestedViewsActivateCallsApp1P2V2);
-					assert.isNull(multipleAndNestedViewsActivateCallsApp1P2S1View.parentNode);
-
-
-
-					// Now multipleAndNestedViewsActivateCallsApp1P2V1View ActivateCallCounts should be 1
-					checkActivateCallCount(multipleAndNestedViewsActivateCallsApp1P2V2View, 1, true);
-					checkActivateCallCount(multipleAndNestedViewsActivateCallsApp1V7View, 2, true);
-					checkActivateCallCount(multipleAndNestedViewsActivateCallsApp1V2View, 1, true);
-					checkActivateCallCount(multipleAndNestedViewsActivateCallsApp1V1View, 3, true);
-					checkActivateCallCount(multipleAndNestedViewsActivateCallsApp1S1View, 4, true);
-					checkActivateCallCount(multipleAndNestedViewsActivateCallsApp1P1View, 4, true);
-
-					// Now multipleAndNestedViewsActivateCallsApp1P2V1View DeactivateCallCounts should be 1
-					checkDeactivateCallCount(multipleAndNestedViewsActivateCallsApp1P2V2View, 0, true);
-					checkDeactivateCallCount(multipleAndNestedViewsActivateCallsApp1V7View, 2, true);
-					checkDeactivateCallCount(multipleAndNestedViewsActivateCallsApp1V1View, 3, true);
-					checkDeactivateCallCount(multipleAndNestedViewsActivateCallsApp1V2View, 1, true);
-					checkDeactivateCallCount(multipleAndNestedViewsActivateCallsApp1S1View, 4, true);
-					checkDeactivateCallCount(multipleAndNestedViewsActivateCallsApp1P1View, 4, true);
+			return when(new Application(JSON.parse(stripComments(multipleAndNestedViewsActivateCallsconfig1)),
+					multipleAndNestedViewsActivateCallsContainer1).then(function (app) {
+					// we are ready to test
+					testApp = app;
+
+					var multipleAndNestedViewsActivateCallsApp1P1 = document.getElementById("content_P1");
+					var multipleAndNestedViewsActivateCallsApp1ContentView =
+						viewUtils.getViewFromViewId(testApp, "content");
+					multipleAndNestedViewsActivateCallsApp1Content =
+						multipleAndNestedViewsActivateCallsApp1ContentView.containerNode;
+
+					// Here multipleAndNestedViewsActivateCallsApp1Home1View should be displayed
+
+					multipleAndNestedViewsActivateCallsApp1P1View = viewUtils.getViewFromViewId(testApp, "content_P1");
+					// check the DOM state to see if we are in the expected state
+					assert.isNotNull(multipleAndNestedViewsActivateCallsNode1,
+						"root multipleAndNestedViewsActivateCallsNode1 must be here");
+					assert.isNotNull(multipleAndNestedViewsActivateCallsApp1P1,
+						"multipleAndNestedViewsActivateCallsApp1Home1 view must be here");
+				}))
+				.then(function () {
+					// Currently showing P1_S1_V1 test transition to V7
+					return when(multipleAndNestedViewsActivateCallsApp1Content.show("V7").then(function () {
```

Comment:
this can presumably be unwound, instead of

```js
.then(function () { return when(multipleAndNestedViewsActivateCallsApp1Content.show("V7")).then( ... ); } )
```

do

```js
.then(function () { return multipleAndNestedViewsActivateCallsApp1Content.show("V7"); })
.then(...)
```